From a934d0a31febca11d70bdeb9ec848ebc5760779b Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 5 Mar 2018 10:49:52 +0200
Subject: [PATCH 1255/1345] fix: net: pp2x: fix panic on mtu change to from
 jumbo

commit  2ee77ea5bae4c89e0853c75ff8c99225facf983f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Symptoms:
- Bad address panic on kfree or mv_pp2x_bm_bufs_free
- Bad address panic on rx-polling

Root-cause:
old_short_pool is used uninitialized

FIX:
Initialize the old_short_pool with the old_short_port_pool->log_id

Change-Id: I6d57e382fdf33b5e7badfb895b53820636fc6508
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/51301
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 6005acd..27d5b7c 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -759,6 +759,7 @@ static int mv_pp2x_bm_update_mtu(struct net_device *dev, int mtu)
 	int pkt_size = MVPP2_RX_PKT_SIZE(mtu);
 
 	old_long_pool = old_long_port_pool->log_id;
+	old_short_pool = old_short_port_pool->log_id;
 
 	/* If port MTU is higher than 1518B:
 	* HW Long pool - SW Jumbo pool, HW Short pool - SW Short pool
-- 
1.7.9.5

